src/hs_ontology_api/routes/fieldassays/fieldassays_controller.py

```diff
@@ -0,0 +1,58 @@
+# coding: utf-8
+# JAS December 2023
+from flask import Blueprint, jsonify, current_app, request, make_response
+from hs_ontology_api.utils.neo4j_logic import field_assays_get_logic
+from hs_ontology_api.utils.field_error_string import get_error_string
+
+field_assays_blueprint = Blueprint('field-assays', __name__, url_prefix='/field-assays')
+
+
+def field_assays_get(name=None):
+    """Returns detailed information on associations between metadata fields and assay
+        :param name: field name (used in the field-assays/<name> route)
+        :rtype: Union[List[FieldAssay]]
+
+    """
+
+    # Get optional filtering parameters:
+    # assay_identifier - legacy identifier for assay dataset from field_assays.yaml.
+    #                    Can be a data_type, alt-name, or description.
+    # data_type - legacy data_type, used in ingestion workflows
+    # dataset_type - "soft assay" dataset type, used by the Rules Engine
+
+    for req in request.args:
+        if req not in ['assay_identifier', 'data_type', 'dataset_type']:
+            return make_response(f"Invalid parameter: '{req}'", 400)
```

Comment:
Instead of just telling the users the parameter is invalid, it's more helpful to show the ones that can be used to avoid guess work or a trip to look up the Smart API specs.

Reply:
done